examples/dataset_generation/create_alpaca_dataset.ipynb

```diff
@@ -0,0 +1,176 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generate dataset for ALPACA\n",
+    "\n",
+    "### Steps\n",
+    "\n",
+    "1. Bootstrap the instructions using\n",
+    "2. Classify instructions if they are of type classification or not using\n",
+    "3. Generate instances using\n",
+    "4. Filter out low-quality and repeated instructions using\n",
+    "\n",
+    "## But we will do all this steps for you!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "WARNING: CUDA is not available, using CPU instead, can be very slow\n"
+     ]
+    }
+   ],
+   "source": [
+    "from xturing.datasets import InstructionDataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "./self_instruct_davinci_cache_10_5 directory already exists. Please specify other directory if you don't want to use previous cache.\n",
+      "Loaded 15 machine-generated instructions from file.\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "15it [00:00, 25035.64it/s]           \n",
+      "10it [00:00, 22180.35it/s]\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Loaded 10 existing requests\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 10/10 [00:00<00:00, 1863.56it/s]\n",
+      "10it [00:00, 1397.03it/s]\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Loaded 10 existing requests\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 10/10 [00:00<00:00, 753.37it/s]\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Loaded 10 raw generated tasks\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 10/10 [00:00<00:00, 30660.12it/s]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Saved 15 instances\n",
+      "Unique instructions: 7\n",
+      "Classification instructions: 0\n",
+      "Non-classification instructions: 7\n",
+      "Sampling 5 instructions\n",
+      "Only using 13 instances for these sampled instructions.\n",
+      "Included 175 seed tasks\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "instruction_dataset = InstructionDataset.generate_dataset(api_key=\"<your api key>\", path=\"./seed_tasks.jsonl\")"
```

Comment:
How does the user add organization? 

Also, how to use any other model except davinci?

Reply:
Should we instead create an OpenAI models class from which you can choose? davinci, ada, etc.?

Reply:
engine argument is supported, so user can provide different model from OpenAI
also organization argument is supported, but for standard api key it is not needed
Added information how to set this parameters to .ipynb